### ISSUE TITLE
Error when including class and missing a colon

### ISSUE THREAD

**Issue by peelman** [rodjek/puppet-lint#507]
Getting an error when including classes...

Using Atom.io with Linter 1.11.14, linter-puppet-lint 0.7.0, and puppet-lint 2.0.0.
#### fails

```
include users:banished
```
#### succeeds

```
include users::banished
```
#### Error Message

```
Error: /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/plugins/check_whitespace.rb:127:in `block (2 levels) in check': undefined method `each' for nil:NilClass (NoMethodError)
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/plugins/check_whitespace.rb:111:in `each'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/plugins/check_whitespace.rb:111:in `each_with_index'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/plugins/check_whitespace.rb:111:in `block in check'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/plugins/check_whitespace.rb:95:in `each'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/plugins/check_whitespace.rb:95:in `check'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/checkplugin.rb:21:in `run'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/checks.rb:58:in `block in run'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/checks.rb:56:in `each'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/checks.rb:56:in `run'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint.rb:174:in `run'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/bin.rb:59:in `block in run'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/bin.rb:56:in `each'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/lib/puppet-lint/bin.rb:56:in `run'
    from /Users/npeelman/.rbenv/versions/2.2.3/lib/ruby/gems/2.2.0/gems/puppet-lint-2.0.0/bin/puppet-lint:7:in `<top (required)>'
    from /Users/npeelman/.rbenv/versions/2.2.3/bin/puppet-lint:23:in `load'
    from /Users/npeelman/.rbenv/versions/2.2.3/bin/puppet-lint:23:in `<main>'
    at ChildProcess.<anonymous> (/Users/npeelman/Dropbox/.atom/packages/linter-erb/node_modules/atom-linter/node_modules/sb-exec/lib/index.js:36:20)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at maybeClose (internal/child_process.js:818:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
```

Ask if you need anything else or have trouble replicating...

**Comment by ghoneycutt** [comment]
This is a syntax issue with the manifest, not a style issue.

**Comment by peelman** [comment]
Correct.  But is the script-equivalent of a crash a desired behavior?  I can do all sorts of other syntactically-stupid things without causing a similar errors...

**Comment by mschuchard** [comment]
linter-puppet-lint maintainer here wandering by. This looks like a legitimate error with puppet lint. If you look at the declaration of `level_tokens` here:

https://github.com/rodjek/puppet-lint/blob/master/lib/puppet-lint/plugins/check_whitespace.rb#L98

it is never assigned a value prior to being iterated over in this conditional block here:

https://github.com/rodjek/puppet-lint/blob/master/lib/puppet-lint/plugins/check_whitespace.rb#L127

which means any logic flow that hits that conditional will fail. Also I am unsure how a colon is being interpreted as a right brace or semicolon.

On another note, the stack trace has:

at ChildProcess.<anonymous> (/Users/npeelman/Dropbox/.atom/packages/linter-erb/node_modules/atom-linter/node_modules/sb-exec/lib/index.js:36:20)

unsure why it is that the apm for linter-puppet-lint is using the sb-exec from linter-erb. Seems like something funky with the atom-linter package is happening there.

**Comment by rnelson0** [comment]
We can look at addressing the crash, but ghoneycutt is correct that puppet-lint does not validate the code, only lints it, so the syntax error will not be addressed.

Random thought: should `puppet parser validate` be run against the code to be tested first, before linting it?

**Comment by mschuchard** [comment]
> Random thought: should puppet parser validate be run against the code to be tested first, before linting it?

Since the purpose of this software is to check style, and adding a syntax check would add the large dependency that is puppet for the `puppet::face` api, I would guess that reaction to that kind of move would be negative.

**Comment by rnelson0** [comment]
We are likely to be leveraging the puppet gem for the parser and lexer in 3.0.0. This will be addressed in #586 to prevent the crashes for invalid syntax.

**Comment by rodjek** [comment]
This syntax error is now handled gracefully by #809
